src/test/java/org/jitsi/meet/test/pageobjects/web/ToolbarV2.java

```diff
@@ -0,0 +1,416 @@
+     * popup.
+     */
+    public void clickFilmstripOnlySettingsButton()
+    {
+        clickButton(SETTINGS);
+    }
+
+    /**
+     * Clicks on the hangup toolbar button which leaves the current conference.
+     */
+    public void clickHangUpButton()
+    {
+        attemptToClickButton(HANGUP);
+    }
+
+    /**
+     * Clicks on the info toolbar button which opens or closes the info dialog.
+     */
+    public void clickInfoButton()
+    {
+        clickButton(INFO);
+    }
+
+    /**
+     * Clicks on the overflow toolbar button which opens or closes the overflow
+     * menu.
+     */
+    public void clickOverflowButton() {
+        clickButton(OVERFLOW);
+    }
+
+    /**
+     * Clicks on the profile toolbar button which opens or closes the profile
+     * panel.
+     */
+    public void clickProfileButton()
+    {
+        clickButtonInOverflowMenu(PROFILE);
+    }
+
+    /**
+     * Clicks on the recording toolbar button which toggles recording or live
+     * streaming.
+     */
+    public void clickRecordButton()
+    {
+        clickButtonInOverflowMenu(RECORD);
+    }
+
+    /**
+     * Clicks on the settings toolbar button which opens or closes the settings
+     * panel.
+     */
+    public void clickSettingsButton()
+    {
+        clickButtonInOverflowMenu(SETTINGS);
+    }
+
+    /**
+     * Clicks on the shared video toolbar button which toggles sharing of a
+     * YouTube video.
+     */
+    public void clickSharedVideoButton()
+    {
+        clickButtonInOverflowMenu(SHARE_VIDEO);
+    }
+
+    /**
+     * Clicks on the video mute toolbar button which toggles video mute.
+     */
+    public void clickVideoMuteButton()
+    {
+        clickButton(VIDEO_MUTE);
+    }
+
+    /**
+     * Clicks on the video quality toolbar button which opens or closes the
+     * dialog for adjusting max-received video quality.
+     */
+    public void clickVideoQualityButton()
+    {
+        clickButtonInOverflowMenu(VIDEO_QUALITY);
+    }
+
+    /**
+     * Ensures the overflow menu is not displayed.
+     */
+    public void closeOverflowMenu()
+    {
+        if (!isOverflowMenuOpen())
+        {
+            return;
+        }
+
+        clickOverflowButton();
+    }
+
+    /**
+     * Gets the participant's avatar image element located in the toolbar.
+     *
+     * @return <tt>WebElement</tt> for the toolbar profile image.
+     */
+    public WebElement getProfileImage()
+    {
+        openOverflowMenu();
+
+        return participant.getDriver().findElement(By.cssSelector(
+            getAccessibilityCSSSelector(PROFILE) + " img"));
+    }
+
+    /**
+     * Checks whether or not the recording button is present in the toolbar.
+     *
+     * @return True if the recording button is present, false if it is not.
+     */
+    public boolean hasRecordButton()
+    {
+        openOverflowMenu();
+
+        List<WebElement> elements
+            = participant.getDriver().findElements(By.cssSelector(
```

Comment:
this is the only `hasXXXButton` method i assume? that's why there isn't a generic `hasButton(String label)` method?

Reply:
Hm, very likely but not intentionally. I was breaking out helpers when I saw some re-use potential, and it looks like only JibriTest uses a hasXXXButton method. It seems like all the other tests assume the button will be present.